### PR TITLE
[timeseries] fix random seed, reset seed for every model

### PR DESCRIPTION
**Pull request by canerturkmen**
*Issue #, if available:*

#3896 
#3923

*Description of changes:*

Previously, `seed_everything` was called once at the beginning of `TimeSeriesPredictor` `fit` and `predict` methods which led to two undesirable side effects:
- if any of the models reset the seeds in their internal logic, they could effectively cancel the `random_seed` provided by the user. This was done, e.g., in `RecursiveTabular`'s `NN_TORCH` affecting `high_quality` presets.
- The fitting order of models could affect the outcomes of individual models. i.e., the random state at the beginning of each model depended on previous models.

With this change, we propagate the random seed to the learner and trainer both at training and prediction time. We set the seed again, per model, when fitting single models or predicting with single models.

Also fixes a minor issue in predictor unit tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: timeseries/tests/unittests/test_predictor.py [hunk 1]
```diff
@@ -1087,3 +1095,66 @@ def test_when_predictor_fit_with_verbosity_then_verbosity_overridden_and_propaga
...
+
+    def train_save_side_effect(self, *args, **kwargs):
+        assert torch.get_rng_state().numpy()[0] == random_seed
+
+        # mess with the seed
+        seed_everything(66)
+
+        return "mock_model"
+
+    with mock.patch("autogluon.timeseries.trainer.AbstractTimeSeriesTrainer._train_and_save") as mock_train_save:
```
Comment (shchur) [inline timeseries/tests/unittests/test_predictor.py:1114]: Nice!

File: timeseries/tests/unittests/test_predictor.py [hunk 2]
```diff
@@ -1034,26 +1036,32 @@ def test_when_log_to_file_set_then_predictor_logs_to_file(temp_model_path):
...
-    # check if the log contains text
-    with open(log_path, "r") as f:
-        log_text = f.read()
-    assert "Naive" in log_text
+        # check if the log contains text
+        with open(log_path, "r") as f:
+            log_text = f.read()
+        assert "Naive" in log_text
+    finally:
+        log_path.unlink(missing_ok=True)
```
Comment (shchur) [inline timeseries/tests/unittests/test_predictor.py:1064]: Why is this change necessary?

Comment (canerturkmen) [reply timeseries/tests/unittests/test_predictor.py:1064]: The test left the `custom_log.txt` file in the local folder. This makes the test cleans up after itself.

**Review by shchur — APPROVED** [review]
LGTM! Feel free to merge once the tests pass.
